ruby/src/IceRuby/Util.cpp

```diff
@@ -677,41 +545,105 @@ IceRuby::createArray(long sz)
     return arr;
 }
 
-VALUE
-IceRuby::convertLocalException(std::exception_ptr eptr)
+namespace
 {
-    //
-    // We cannot throw a C++ exception or raise a Ruby exception. If an error
-    // occurs while we are converting the exception, we do our best to return
-    // an appropriate Ruby exception.
-    //
-    try
-    {
-        rethrow_exception(eptr);
-    }
-    catch (const Ice::LocalException& ex)
+    template<size_t N> VALUE createRubyException(
+        const char* typeId,
+        std::array<VALUE, N> args,
+        bool fallbackToLocalException = false)
     {
+        string className = string{typeId}.substr(2);
+        VALUE rubyClass = T_NIL;
         try
         {
-            string name = string{ex.ice_id()}.substr(2);
-            volatile VALUE cls = callRuby(rb_path2class, name.c_str());
-            if (NIL_P(cls))
```

Comment:
I believe this code was bogus. If callRuby fails, it throws RubyException - it doesn't return a nil value.